From 9a4fb170134abcdfcc696c98b3b0160815a20be6 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Fri, 24 Aug 2018 15:27:20 -0700
Subject: [PATCH] link: drop the forced use of gold on arm, aarch64

as the original issue with BFD has long been fixed,
and using gold on aarch64 is causing other issues.

Signed-off-by: Matt Madison <matt@madison.systems>
---
 src/cmd/link/internal/ld/lib.go | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/src/cmd/link/internal/ld/lib.go b/src/cmd/link/internal/ld/lib.go
index 11cbb8a8bb..d8d9421ea2 100644
--- a/src/cmd/link/internal/ld/lib.go
+++ b/src/cmd/link/internal/ld/lib.go
@@ -1170,28 +1170,6 @@ func (ctxt *Link) hostlink() {
 		// can move symbols out of sections that rely on stable offsets
 		// from the beginning of the section (like sym.STYPE).
 		argv = append(argv, "-Wl,-znocopyreloc")
-
-		if ctxt.Arch.InFamily(sys.ARM, sys.ARM64) {
-			// On ARM, the GNU linker will generate COPY relocations
-			// even with -znocopyreloc set.
-			// https://sourceware.org/bugzilla/show_bug.cgi?id=19962
-			//
-			// On ARM64, the GNU linker will fail instead of
-			// generating COPY relocations.
-			//
-			// In both cases, switch to gold.
-			argv = append(argv, "-fuse-ld=gold")
-
-			// If gold is not installed, gcc will silently switch
-			// back to ld.bfd. So we parse the version information
-			// and provide a useful error if gold is missing.
-			cmd := exec.Command(*flagExtld, "-fuse-ld=gold", "-Wl,--version")
-			if out, err := cmd.CombinedOutput(); err == nil {
-				if !bytes.Contains(out, []byte("GNU gold")) {
-					log.Fatalf("ARM external linker must be gold (issue #15696), but is not: %s", out)
-				}
-			}
-		}
 	}
 
 	if ctxt.IsELF && len(buildinfo) > 0 {
-- 
2.17.1

